packages/picasso/src/CalendarMonthHeader/CalendarMonthHeader.tsx

```diff
@@ -1,36 +1,124 @@
-import React from 'react'
-import { format } from 'date-fns'
+import type { ReactNode } from 'react'
+import React, { useContext } from 'react'
+import type { CaptionProps } from 'react-day-picker'
+import { useDayPicker, useNavigation } from 'react-day-picker'
+import isSameMonth from 'date-fns/isSameMonth'
 
 import ButtonCircular from '../ButtonCircular'
 import Typography from '../Typography'
 import { ChevronMinor24, BackMinor24 } from '../Icon'
-import type { MonthHeaderProps } from '../Calendar/types'
 import Container from '../Container'
+import CalendarContext from '../CalendarContext'
 
-const CalendarMonthHeader = ({
-  switchMonth,
-  activeMonth: headerActiveMonth,
-}: MonthHeaderProps) => (
-  <Container flex justifyContent='space-between' bottom='medium'>
-    <ButtonCircular
-      title='Previous month'
-      aria-label='Previous month'
-      variant='flat'
-      icon={<BackMinor24 />}
-      onClick={() => switchMonth(-1)}
-    />
-
-    <Typography variant='heading' size='medium'>
-      {format(headerActiveMonth, 'MMMM y')}
-    </Typography>
-    <ButtonCircular
-      title='Next month'
-      aria-label='Previous month'
-      variant='flat'
-      icon={<ChevronMinor24 />}
-      onClick={() => switchMonth(1)}
-    />
-  </Container>
-)
+interface RenderMonthHeaderProps extends CaptionProps {
+  /** Unique key */
+  key?: string
+  /** Children nodes */
+  children?: ReactNode
+}
+
+export type RenderMonthHeader = (args: RenderMonthHeaderProps) => ReactNode
+
+const CalendarMonthHeader = (props: CaptionProps) => {
+  const {
+    formatters: { formatCaption },
+    locale,
+    classNames,
+    disableNavigation,
+    styles,
+    numberOfMonths,
+    onMonthChange,
+    dir,
+  } = useDayPicker()
+  const { previousMonth, nextMonth, goToMonth, displayMonths } = useNavigation()
+  const { renderMonthHeader } = useContext(CalendarContext)
+
+  if (disableNavigation) {
+    return (
+      <div className={classNames.caption} style={styles.caption}>
+        <Typography
+          variant='heading'
+          size='medium'
+          aria-live='polite'
+          aria-atomic='true'
+        >
+          {formatCaption(props.displayMonth, { locale })}
+        </Typography>
+      </div>
+    )
+  }
+
+  const displayIndex = displayMonths.findIndex(month =>
+    isSameMonth(props.displayMonth, month)
+  )
+
+  let isFirst = displayIndex === 0
+  let isLast = displayIndex === displayMonths.length - 1
+
+  if (dir === 'rtl') {
+    ;[isLast, isFirst] = [isFirst, isLast]
+  }
+
+  const hideNext = numberOfMonths > 1 && (isFirst || !isLast)
+  const hidePrevious = numberOfMonths > 1 && (isLast || !isFirst)
```

Comment:
Wouldn't this be the same behavior?
```suggestion
  const hideNext = numberOfMonths > 1 && !isLast
  const hidePrevious = numberOfMonths > 1 && !isFirst
```

Reply:
@dmaklygin What do you think about removing the code that is responsible for hiding the month buttons when multiple months are shown from this pull request (but still keeping it as it will be needed in https://toptal-core.atlassian.net/browse/FX-3963)?

Reply:
I think, the `next` button should be hidden if the month is last
the similar logic for the `previous` button: should be hidden if the month is first. 
no?

Reply:
@dmaklygin I think it is like this – we show previous button only for the first month and show next button only for the last month

```
|  < May    |    June    |    July >  |
```

Reply:
But we can still discuss it as part of https://toptal-core.atlassian.net/browse/FX-3963

Reply:
I think, it's out of scope of this task. We will implement it in a specifed one, so let's not focus on it right now. I think, it's a minor thing.